lnwallet/fee_estimator.go

```diff
@@ -518,3 +543,206 @@ func (s SparseConfFeeSource) ParseResponse(r io.Reader) (map[uint32]uint32, erro
 // A compile-time assertion to ensure that SparseConfFeeSource implements the
 // WebAPIFeeSource interface.
 var _ WebAPIFeeSource = (*SparseConfFeeSource)(nil)
+
+// WebAPIFeeEstimator is an implementation of the FeeEstimator interface that
+// queries an HTTP-based fee estimation from an existing web API.
+type WebAPIFeeEstimator struct {
+	started sync.Once
+	stopped sync.Once
+
+	// apiSource is the backing web API source we'll use for our queries.
+	apiSource WebAPIFeeSource
+
+	// updateFeeTimer is the timer responsible for updating the Estimator's
+	// fee estimates every time it fires.
+	updateFeeTimer *time.Timer
```

Comment:
If this uses a `time.Ticker` then it doesn't need to be reset. Going further, it can use our `ticker.Ticker` to make it easier to mock out in unit tests.

Reply:
fixed